### PR TITLE
Added pagination to TemplateList

### PR DESCRIPTION
**Pull request by bentshrimp**
## Describe changes

- All styled components exists in TemplateComponents.js
- pagination added into TemplateList

## Types of changes

What is the type of code change?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (changes that resolve errors)
- [x] New feature (changes which adds functionality)
- [ ] Breaking change (big changes that affect existing functionality)
- [ ] Documentation Update

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the "README.md"
- [ ] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

_Please let me know if there's anything else to explain_

### CODE REVIEW

File: src/components/common/template/TemplateCoponents.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    width: "auto",
+  },
+}));
+
+export const SearchIconWrapper = styled("div")(({ theme }) => ({
+  display: "flex",
+  position: "absolute",
+  height: "100%",
+  padding: theme.spacing(0, 2),
+  alignItems: "center",
```
Comment (ancy0) [inline src/components/common/template/TemplateCoponents.js:34]: 😢

**Review by ancy0 — COMMENTED** [review]
👍

File: src/components/common/template/TemplateCoponents.js (hunk 1 above)
Comment (bentshrimp) [reply src/components/common/template/TemplateCoponents.js:34]: always miss something

**Review by ymw0407 — COMMENTED** [review]
image
****


GOOD!

**Review by ancy0 — APPROVED** [review]
👍

**Pull request merged**